javascript/extractor/lib/typescript/src/ast_extractor.ts

```diff
@@ -202,6 +202,12 @@ export function augmentAst(ast: AugmentedSourceFile, code: string, project: Proj
             }
         }
 
+        // This PR: https://github.com/microsoft/TypeScript/pull/50343/ changed how illegal decorators are stored, this hack reverts it.
+        // Those decorator uses are syntax errors, but people use them anyway.
+        if ((node as any).illegalDecorators) {
+            (node as any).decorators = (node as any).illegalDecorators;
```

Comment:
I'd like to avoid modifying the `decorators` property directly. Injecting our own properties (prefixed with `$`) is one thing, but modifying properties owned by TypeScript compiler is too brittle.

I'd rather to something like `node.decorators ?? node.illegalDecorators` at the point where we access the decorators.